datafusion/common/src/scalar.rs

```diff
@@ -2090,7 +2173,9 @@ impl TryFrom<ScalarValue> for i32 {
     fn try_from(value: ScalarValue) -> Result<Self> {
         match value {
             ScalarValue::Int32(Some(inner_value))
-            | ScalarValue::Date32(Some(inner_value)) => Ok(inner_value),
+            | ScalarValue::Date32(Some(inner_value))
+            | ScalarValue::Time32Second(Some(inner_value))
+            | ScalarValue::Time32Millisecond(Some(inner_value)) => Ok(inner_value),
```

Comment:
Question: why is it necessary to add this to the tryfrom implementation? This seems like it's only needed for some workaround, maybe.

In any case, I think a comment would be helpful here; ideally, let's update the comment above (l. 2169).